devlib/trace/perf.py

```diff
@@ -131,7 +163,25 @@ def _get_target_outfile(self, label):
     def _build_perf_command(self, options, events, label):
         event_string = ' '.join(['-e {}'.format(e) for e in events])
         command = PERF_COMMAND_TEMPLATE.format(self.binary,
+                                               self.perf_mode,
                                                options or '',
                                                event_string,
                                                self._get_target_outfile(label))
         return command
+
+    def _get_target_perfdatafile(self):
+        return self.target.get_workpath('perf.data')
+
+    def _get_target_reportfile(self, label):
+        return self.target.get_workpath('{}.rpt'.format(label))
+
+    def _get_target_tracefile(self, label):
+        return self.target.get_workpath('{}.trace'.format(label))
+
+    def _build_perf_report_command(self, reportoptions, label):
+        reportoptions_string = reportoptions
+        command = PERF_REPORT_COMMAND_TEMPLATE.format(self.binary,
+                                                      reportoptions_string,
+                                                      self._get_target_perfdatafile(),
```

Comment:
When testing this on my linux device the `perf.data` file is generated in the default login location due to the automatic generation rather than the workpath that this expects. If there is no way to control where this file is generated could I suggest explicitly changing to the desired output folder before running the record command to ensure that the output file is in the correct location?

Reply:
On Android it seems to include a `cd` to the output folder as part of the command execution:
`2019-04-29 19:23:04,804 DEBUG       android:         adb -s 10.1.16.10:5555 shell echo 'cd sdcard/devlib-target && /data/local/tmp/bin/busybox nohup /data/local/tmp/bin/perf record -a -e migrations -e cs -e r11 sleep 1000 > /sdcard/devlib-target/perf_0.out 2>&1  &' | su`

I'd assume that it is the same on Linux but haven't confirmed. I can look at running on Linux target but will take me a little time. Do you happen to have the run.log file from your testing to confirm the command used on Linux?

Reply:
Ah yes I see the same behaviour on my android device however on linux it produces: 
```DEBUG           ssh:         (sh -c '/usr/bin/perf record -a -e migrations -e cs sleep 1000 > /home/root/devlib-target/perf_0.out 2>&1 ' 1>/dev/null 2>/dev/null &); __devlib_ec=$?; echo; echo $__devlib_ec```

For some reason the `kick_off` implementation on linux diffs to androids and does not have the initial `cd` command. I'm not sure why this would be the case and it seems like something that should be corrected to be consistent.